src/php/lib/Grpc/BaseStub.php

```diff
@@ -243,6 +243,43 @@ protected function _simpleRequest($method,
         return $call;
     }
 
+    /**
+     *
+     * @param string   $method      The name of the method to call
+     * @param mixed    $argument    The argument to the method
+     * @param callable $deserialize A function that deserializes the response
+     * @param array    $metadata    A metadata map to send to the server
+     *                              (optional)
+     * @param array    $options     An array of options (optional)
+     */
+    protected function _simpleRequestAsync($method,
+                                   $argument,
+                                   $deserialize,
+                                   $callback,
+                                   array $metadata = [],
+                                   array $options = [])
+    {
+        $options['client_async'] = true;
```

Comment:
Another way to do this is reuse the `_simpleRequest` instead of `_simpleRequestAsync`, and `options` is set when the people call `$stub->SayHello($arg, [client_sync => true, call_back=$call_back])`, so there is no need to add one more method(no need to change protoc codegen).
The same to move `$call->callAsync` into `$call->start()` and check `client_sync` there.

Reply:
Agreed about `_simpleRequest` and `_simpleRequestAsync`: They don't really need to be two separated methods.

About merging `callAsync` and `start`, I think that it would result in more complicated code, however.